src/Data/StrMap.purs

```diff
@@ -112,10 +118,20 @@ instance foldableStrMap :: Foldable StrMap where
   foldr f z m = foldr f z (values m)
   foldMap f = foldMap (const f)
 
+instance foldableWithIndex :: FoldableWithIndex String StrMap where
+  foldlWithIndex f = fold (flip f)
+  foldrWithIndex f z m = foldr (uncurry f) z (toArrayWithKey Tuple m)
+  foldMapWithIndex = foldMap
```

Comment:
I meant `foldMap` specifically. I wasn't seeing how the types lined up. Could you please add a test?